src/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php

```diff
@@ -307,15 +307,32 @@ private function getResourcesToWatch(ContainerBuilder $container, array $config)
     {
         $paths = array_unique(array_merge($this->getBundlesResourcesPaths($container, $config), $config['mapping']['paths']));
 
-        // Flex structure (only if nothing specified)
-        $projectDir = $container->getParameter('kernel.project_dir');
-        if (!$paths && is_dir($dir = "$projectDir/config/api_platform")) {
-            $paths = [$dir];
+        // Default paths
+        if (!$paths) {
+            $projectDir = $container->getParameter('kernel.project_dir');
+            if ($this->isConfigEnabled($container, $config['doctrine']) && is_dir($dir = "$projectDir/src/Entity")) {
+                $paths[] = $dir;
+            }
+            if ($this->isConfigEnabled($container, $config['doctrine_mongodb_odm']) && is_dir($dir = "$projectDir/src/Document")) {
+                $paths[] = $dir;
+            }
+
+            // Flex structure
+            if (is_dir($dir = "$projectDir/config/api_platform")) {
+                $paths[] = $dir;
+            }
         }
 
         $resources = ['yml' => [], 'xml' => [], 'dir' => []];
 
         foreach ($paths as $path) {
+            // Replace parameters in path
+            if (preg_match_all('/%([^%]+)%/', $path, $matches)) {
```

Comment:
I don't understand the need for this.
`%kernel.project_dir%` is interpreted if it is a user-defined value, isn't it?